File: src/main/kotlin/solo/codegolf/medium/networkcablingcodesize/NetworkCablingCodeSize.c
#include <stdlib.h>
#include <stdio.h>

int compare(const void *a, const void *b) {
    return *(long long*)a - *(long long*)b;
}

int main() {
    int N;
    scanf("%d", &N);

    long long *x = malloc(N * sizeof(long long));
    long long *y = malloc(N * sizeof(long long));
    long long min_x = 1LL << 60, max_x = -(1LL << 60);

    for (int i = 0; i < N; i++) {
        scanf("%lld %lld", &x[i], &y[i]);
        if (x[i] < min_x) min_x = x[i];
        if (x[i] > max_x) max_x = x[i];
    }

    // 메인 케이블 길이 계산
    long long main_cable_length = max_x - min_x;

    // y좌표를 정렬하여 중앙값 찾기
    qsort(y, N, sizeof(long long), compare);
    long long median_y = N % 2 == 1 ? y[N/2] : y[N/2-1];

    // 각 건물에서 메인 케이블까지의 거리 계산
    long long vertical_length = 0;
    for (int i = 0; i < N; i++) {
        vertical_length += llabs(y[i] - median_y);
    }

    printf("%lld\n", main_cable_length + vertical_length);

    free(x);
    free(y);
    return 0;
}

File: src/main/kotlin/solo/codegolf/medium/networkcablingcodesize/NetworkCablingCodeSize.kt
import java.util.*
import kotlin.math.abs

fun main() {
    val input = Scanner(System.`in`)
    val n = input.nextInt()

    val x = LongArray(n)
    val y = LongArray(n)
    var minX = Long.MAX_VALUE
    var maxX = Long.MIN_VALUE

    // 건물 위치 입력받기
    for (i in 0 until n) {
        x[i] = input.nextLong()
        y[i] = input.nextLong()

        if (x[i] < minX) minX = x[i]
        if (x[i] > maxX) maxX = x[i]
    }

    // 메인 케이블 길이 계산
    val mainCableLength = maxX - minX

    // y좌표 정렬하여 중앙값 찾기
    y.sort()
    val medianY = if (n % 2 == 1) y[n / 2] else y[n / 2 - 1]

    // 각 건물에서 메인 케이블까지의 거리 계산
    var verticalLength = 0L
    for (i in 0 until n) {
        verticalLength += abs(y[i] - medianY)
    }

    // 총 케이블 길이 출력
    println(mainCableLength + verticalLength)
}